front/selector.js

```diff
@@ -0,0 +1,13 @@
+export default function select(data, getList, defaults = {}) {
+  if (data && typeof data === 'object') {
+    const selected = {}
+    for (const d in data) {
```

Comment:
Você tem duas opções:

1. Usar for..in e adotar hasOwnProperty ([por que?](https://stackoverflow.com/questions/8312459/iterate-through-object-properties)).
2. Usar algo como Object.entries num for..of